pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -587,11 +660,11 @@ func TestUpdateNodeWithMultipleTaints(t *testing.T) {
 
 func TestUpdateNodeWithMultiplePods(t *testing.T) {
 	testCases := []struct {
-		description         string
-		pods                []v1.Pod
-		oldNode             *v1.Node
-		newNode             *v1.Node
-		expectedDeleteTimes durationSlice
+		description  string
+		pods         []v1.Pod
+		oldNode      *v1.Node
+		newNode      *v1.Node
+		expectDelete []string
```

Comment:
It seems we might be reducing the test coverage here by not asserting on the delete times.
I suggest using fake.clock to simulate the time passage. For example: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/disruption/disruption_test.go

This will also require modifying the implementation of the taint_manager to use a clock rather than call time directly. For example implementation using a clock check: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/disruption/disruption.go

Reply:
Oh, ok. Now the coverage is 90%, but, I think it will increase it

Reply:
Don't you think, that it's better to make a new PR for such task?

Reply:
I think we should avoid making temporary regressions in coverage. Even if it is not regression in the sense of % it is sort of conceptual regression as we don't assert on the timings. 

Thus, I suggest either:
- revert the unnecessary simplification in this PR and use the fake clock in a followup PR or,
- use a fake clock as part of this PR. 

The use of fake clock would also eliminate the waiting functions used to assert in the other places.

Reply:
It seems, that I don't understand it quete well. In disruption.go fake clock is not used for concurrency. Then, how it can be applicable for taint_manager.go?